### ISSUE TITLE
Having multiple GhTokenWorkflow tokens results in error when running action Update AL-Go System Files

### ISSUE THREAD

**Issue by louagej** [microsoft/AL-Go#878]
Hi,

I'm working for multiple companies that are using AL-GO Template repositories (2 different github organisations).
In order to separated secrets between the 2 companies, I've create 2 tokens for running actions in the different companies:

![image](https://github.com/microsoft/AL-Go/assets/7229592/f07093a7-f6aa-4ad4-b94c-334793057d38)

The repository that uses the Fine-grained token `GhTokenWorkflow` is able to update the AL-GO system files
The repository that uses the Fine-grained token `GhTokenWorkflow Company B` is not able to update the AL-GO system files.
Both Tokens have following permissions:

![image](https://github.com/microsoft/AL-Go/assets/7229592/4c00a0a4-ae52-4c48-b95b-1a2f3c4ee816)

Both repositories have a secret called `GhTokenWorkflow`

In the repository where both Token and Secret is called  `GhTokenWorkflow`, the action runs fine.

But In the repository where the Token is called `GhTokenWorkflow Company B` and Secret is called  `GhTokenWorkflow`, the action returns following Error:

**UpdateALGoSystemFiles**
```
Unexpected error when running action.
Error Message: Failed to update AL-Go System Files.
Make sure that the personal access token, defined in the secret called GhTokenWorkflow, is not expired and it has permission to update workflows.
(Error was Cloning into 'Repo-A'... remote: Write access to repository not granted. fatal: unable to access 'https://github.com/Company B/Repo-A/':
    The requested URL returned error: 403
    ExitCode: 128 Commandline: git clone ***github.com/Company B/Repo-A ),
       StackTrace: at <ScriptBlock>, D:\a\_actions\microsoft\AL-Go-Actions\v4.0\CheckForUpdates\CheckForUpdates.ps1: line 523 <- at <ScriptBlock>, D:\a\_temp\1b9f4e6c-6324-42cc-8fb5-7018f13ea052.ps1: line 4 <- at <ScriptBlock>, <No file>: line 1
```

**Comment by freddydk** [comment]
Fine grained tokens are created for specific repositories:
![image](https://github.com/microsoft/AL-Go/assets/10775043/fbad7ab7-5df8-4a4a-b9d8-3a2be6e5fa5c)

What did you select there?
You can modify this without re-issuing the token.

**Comment by louagej** [comment]
I selected All Repositories.

**Comment by freddydk** [comment]
There is definitely no problem in having multiple tokens and then using them as secrets in different repositories - I have that in multiple accounts.
The problem here must be a copy/paste error or a permission error of some kind.
Did you select Company B as resource owner for 2nd token?
Did you assign the following permissions to the repositories:
- Contents: Read and Write
- Metadata: Read only
- Pull Requests: Read and Write
- Workflows: Read and Write
(You can edit permissions and keep the same token)

**Comment by louagej** [comment]
Hi Freddy,

I did select Company B as owner of the token.

In order to make sure there is no copy/paste error, I regenerated the token and updated the secret in the repository.

But the problem wasn't one of the above, but in the permissions.
I based my permissions on the settings page in https://freddydk.github.io/AL-Go/Scenarios/settings.html.
There is mentioned the token should only have `Workflows: Read and Write` permissions.

So I added the 4 permissions above and I was able to run the action:
- `Contents: Read and Write`
- `Metadata: Read only`
- `Pull Requests: Read and Write`
- `Workflows: Read and Write`

![image](https://github.com/microsoft/AL-Go/assets/7229592/bee8fcff-5f1f-41c1-939e-321104d135ce)

Thank you for your help!
